packages/axelar-wasm-std/src/token.rs

```diff
@@ -0,0 +1,23 @@
+use cosmwasm_std::{Coin, MessageInfo};
+use error_stack::{bail, Result};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("invalid token: one and only one token is required for this operation, got {0:?}")]
+    MultipleTokens(Vec<Coin>),
+}
+
+pub trait GetToken {
```

Comment:
consider
```
mod sealed {
   pub trait Sealed {}
}

pub trait GetToken: sealed::Sealed{
...
}

impl sealed::Sealed for MessageInfo {}

...
```
that way, no other struct can implement the `GetToken` trait

Reply:
Not sure if I wanna prevent it this hard.